src/wallet/rpcsigner.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2018-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparamsbase.h>
+#include <key_io.h>
+#include <rpc/server.h>
+#include <rpc/util.h>
+#include <util/strencodings.h>
+#include <wallet/rpcsigner.h>
+#include <wallet/rpcwallet.h>
+#include <wallet/wallet.h>
+
+#ifdef ENABLE_EXTERNAL_SIGNER
+
+static RPCHelpMan enumeratesigners()
+{
+    return RPCHelpMan{
+        "enumeratesigners",
+        "Returns a list of external signers from -signer.",
+        {},
+        RPCResult{
+            RPCResult::Type::OBJ, "", "",
+            {
+                {RPCResult::Type::ARR, "signers", /* optional */ false, "",
+                {
+                    {RPCResult::Type::STR_HEX, "masterkeyfingerprint", "Master key fingerprint"},
+                    {RPCResult::Type::STR, "name", "Device name"},
+                },
+                }
+            }
+        },
+        RPCExamples{""},
+        [](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
+            std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
```

Comment:
why does this rpc require a wallet?

Reply:
I might get rid of that requirement in a followup.